src/common/utils/resolve_reference.py

```diff
@@ -237,20 +236,19 @@ class ResolvedReference:
     entity: dict | list
 
 
-def resolve_reference(reference: str, get_data_source: Callable) -> ResolvedReference:
+def resolve_reference(address: Address, get_data_source: Callable) -> ResolvedReference:
```

Comment:
As far as I understand, we should make an effort to substitute all function parameters"reference: string" with "address: Address" in this file.

There are still some functions in this file that take in a reference string, for example
 * _reference_to_reference_items()
 * reference_to_reference_items()
 * _next_reference_part()
 * split_reference()
 
 
 since this PR is already very big, it might be better to update these afterwards

Reply:
I agree. I've linked to this message in #389